examples/impute/plot_multiple_imputation.py

```diff
@@ -0,0 +1,392 @@
+    sigma_hat_squared = sum_sq_errors / (len(y_true) - 2)
+    X_prime_X = np.dot(X.T, X)
+    covariance_matrix = sigma_hat_squared / X_prime_X
+    vars = np.diag(covariance_matrix)
+
+    return vars
+
+
+# Apply Rubin's pooling rules as follows.
+# The value of every estimate is the mean of the estimates in each of the m
+# datasets (Qbar). The variance of these estimates is a combination of the
+# variance of each of the m estimates (Ubar) and the variance between the m
+# estimates (B).
+#
+# Make a function that calculates Qbar from m estimates
+def pool_coefs(m_estimates):
+    m = len(m_estimates)
+    Qbar = np.sum(m_estimates, axis=0) / m
+    return Qbar
+
+
+# Make a function that calculates T from m estimates and their variances
+def pool_variances(m_estimates, m_variances, Qbar):
+    m = len(m_estimates)
+    Ubar = np.sum(m_variances, axis=0) / m
+    B = np.sum((Qbar - m_estimates) ** 2, axis=0) / (m - 1)
+    T = Ubar + B + (B/m)
+
+    return T
+
+
+###############################################################################
+
+# EXAMPLE 1. COMPARE STATISTICAL ESTIMATES AND THEIR VARIANCE USING MULTIPLE
+# IMPUTATION IN A LINEAR REGRESSION MODEL.
+
+###############################################################################
+
+
+def get_results_full_dataset(X, y):
+    # Perform linear regression on full data as a way of comparison
+    estimator = RidgeCV()
+    estimator.fit(X, y)
+    y_predict = estimator.predict(X)
+
+    # Save the beta estimates, the variance of these estimates and 1.96 *
+    # standard error of the estimates. The latter is useful to know the 95%
+    # confidence interval.
+    full_coefs = estimator.coef_
+    full_vars = calculate_variance_of_beta_estimates(y, y_predict, X)
+
+    return full_coefs, full_vars
+
+
+def get_results_chained_imputation(X_incomplete, y, random_state=0):
+    # Impute incomplete data with IterativeImputer using single imputation
+    # We perform N_ITER imputations and only use the last imputation.
+    imputer = IterativeImputer(n_iter=N_ITER,
+                               sample_posterior=True,
+                               random_state=random_state)
+    imputer.fit(X_incomplete)
+    X_imputed = imputer.transform(X_incomplete)
+
+    # Perform linear regression on chained single imputed data
+    # Estimate beta estimates and their variances
+    estimator = RidgeCV()
+    estimator.fit(X_imputed, y)
+    y_predict = estimator.predict(X_imputed)
+
+    # Save the beta estimates, the variance of these estimates and 1.96 *
+    # standard error of the estimates
+    chained_coefs = estimator.coef_
+    chained_vars = calculate_variance_of_beta_estimates(
+            y, y_predict, X_imputed)
+
+    return chained_coefs, chained_vars
+
+
+def get_results_mice_imputation(X_incomplete, y, m=5):
+    # Perform a model on each of the m imputed datasets
+    # Estimate the estimates for each model/dataset
+    m_coefs = []
```

Comment:
You should be able to make a comprehension here
```python
m_coefs, m_vars = zip(*[get_results_chained_imputation(...) for i in range(m)])
```

Reply:
Eh it's simple enough. I don't think this will save any compute.

Reply:
I disagree. List comprehensions are pythonic and even if this is difficult to understand at first, this is a recurrent pattern used by Python programmers a lot. I think that our example needs to use good programming practice. However, if it requires a comment explaining what it does, I am not opposed to that.

However, this is a personal opinion, so I don't want to force it on the review. I'll let another reviewer decide which syntax would be the best for the example.

Reply:
Ok, I'm convinced =)

Reply:
I agree with @glemaitre regarding the list comprehensions